internal/services/storagemover/storage_mover_agent_resource.go

```diff
@@ -0,0 +1,227 @@
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := agents.Agent{
+				Properties: agents.AgentProperties{
+					ArcResourceId: model.ArcVirtualMachineId,
+					ArcVMUuid:     model.ArcVmUuid,
+				},
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r StorageMoverAgentResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StorageMover.AgentsClient
+
+			id, err := agents.ParseAgentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model StorageMoverAgentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				properties.Properties.Description = &model.Description
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r StorageMoverAgentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StorageMover.AgentsClient
+
+			id, err := agents.ParseAgentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := StorageMoverAgentResourceModel{
+				Name:           id.AgentName,
+				StorageMoverId: storagemovers.NewStorageMoverID(id.SubscriptionId, id.ResourceGroupName, id.StorageMoverName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				state.ArcVmUuid = model.Properties.ArcVMUuid
+				state.ArcVirtualMachineId = model.Properties.ArcResourceId
+
+				des := ""
+				if model.Properties.Description != nil {
+					des = *model.Properties.Description
+				}
+				state.Description = des
```

Comment:
We can use the functions from the pointer package to condense this to
```suggestion
				state.Description = pointer.From(model.Properties.Description)
```

Reply:
Fixed.